jsoniter-scala-macros/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/macros/JsonCodecMaker.scala

```diff
@@ -931,8 +938,12 @@ object JsonCodecMaker {
           if (discriminator.isEmpty) names
           else names :+ n
         })
-        val required = classInfo.fields.collect {
-          case f if !f.symbol.isParamWithDefault && !isContainer(f.resolvedTpe) => f.mappedName
+        val required: Seq[String] = classInfo.fields.flatMap { f =>
```

Comment:
Are you doing a deep check here? I was wondering would this logic behave in types such as `List[Option[Int]]` or `List[Set[Int]]`, would it force those fields to be present for the nested collections too?

Reply:
No deep checks - type parameters for arrays and iterables are not analyzed, so logic for working with the flag is the same for any kind of them, but logic of the flag work for all such kind of fields for nested data structures.

I've added more tests for codecs derived with `requireCollectionFields = true` to answer you questions. Please check if something still not clear about the flag.